components/director/internal/tracking/request.go

```diff
@@ -0,0 +1,85 @@
+package tracking
```

Comment:
In general this file seems like doing the same as the different flows for authenticating JWT, Cert and One-time-token. 

Perhaps it would be better to split the *trackingService* in 3 different calls which are accordingly to each responsible component. For certificate it would be connector, for JWT and OTT it would be director. This will also remove the dependency of director to connector. 

The tracking service can put the client identification in the header/body and then it can be inspected in director (as the last component in the chain) and used in prometheus metric.

Reply:
We discussed offline that it would be better to instrument client calls directly into the Tenant Mapping mutator right before we many any DB calls.